File: source/code-snippets/usage-examples/distinct.js
import { MongoClient } from "mongodb";

// Replace the uri string with your MongoDB deployment's connection string.
const uri = "<connection string uri>";

const client = new MongoClient(uri);

async function run() {
  try {
    // define a database and collection on which to run the method
    const database = client.db("sample_mflix");
    const movies = database.collection("movies");

    // specify the document field
    const fieldName = "year";

    // specify an optional query document
    const query = { directors: "Barbra Streisand" };

    const distinctValues = await movies.distinct(fieldName, query);

    console.log(distinctValues);
  } finally {
    await client.close();
  }
}
run().catch(console.dir);

File: source/includes/crud/Distinct.java
// Retrieves distinct values of a field by using the Java driver

package usage.examples;

import org.bson.Document;

import com.mongodb.MongoException;
import com.mongodb.client.DistinctIterable;
import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoClients;
import com.mongodb.client.MongoCollection;
import com.mongodb.client.MongoCursor;
import com.mongodb.client.MongoDatabase;
import com.mongodb.client.model.Filters;

public class Distinct {
    public static void main(String[] args) {
        // Replace the uri string with your MongoDB deployment's connection string
        String uri = "<connection string uri>";

        try (MongoClient mongoClient = MongoClients.create(uri)) {
            MongoDatabase database = mongoClient.getDatabase("sample_mflix");
            MongoCollection<Document> collection = database.getCollection("movies");

            // Retrieves the distinct values of the "year" field present in documents that match the filter
            DistinctIterable<Integer> docs = collection.distinct("year", Filters.eq("directors", "Carl Franklin"), Integer.class);
            MongoCursor<Integer> results = docs.iterator();

            // Prints the distinct "year" values
            while(results.hasNext()) {
                System.out.println(results.next());
            }
        }
    }
}

